### PR TITLE
840 - Add load metadata command

### PR DESCRIPTION
**Pull request by avrohomgottlieb**
## Issue Number

https://github.com/AlexsLemonade/scpca-portal/issues/840

## Purpose/Implementation Notes

Added `load_metadata` command which takes the `load_data` command and removes all computed file generation functionality.

Added `test_load_metadata` test module which takes the test cases from `test_load_data` and removes all testing related to computed file generation.

## Types of changes

What types of changes does your code introduce?

- New feature (non-breaking change which adds functionality)

## Functional tests

Added `test_load_metadata` test module.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots

N/A

### CODE REVIEW

File: api/scpca_portal/test/management/commands/test_load_metadata.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        self.assertProjectData(project)
+        self.assertFalse(project.has_cite_seq_data)
+        self.assertTrue(project.includes_anndata)
+        self.assertTrue(project.modalities)
+        self.assertEqual(project.multiplexed_sample_count, 0)
+        self.assertEqual(project.organisms, ["Homo sapiens"])
+        # This project contains 3 samples
+        single_cell = 2
+        spatial = 1
+        bulk = 1
```
Comment (avrohomgottlieb) [inline api/scpca_portal/test/management/commands/test_load_metadata.py:134]: What's the deal with this `bulk` sample? It has scpca_id `SCPCS999994`, it's included in the `samples_metadata.csv` file but has to corresponding `json` file.

File: api/scpca_portal/test/management/commands/test_load_metadata.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            "submitter_id",
+            "tissue_ontology_term_id",
+            "WHO_grade",
+        ]
+        self.assertEqual(
+            expected_additional_metadata_keys, project.additional_metadata_keys.split(", ")
+        )
+        self.assertEqual(
+            set(expected_additional_metadata_keys), set(sample.additional_metadata.keys())
+        )
```
Comment (avrohomgottlieb) [inline api/scpca_portal/test/management/commands/test_load_metadata.py:172]: Based on previous conversations, I feel like we'd want to delete these metadata key assertion tests (also in `test_spatial_metadata` and `test_multiplexed_metadata`). At some point soon we're going to invert this logic in any case, but in general we discussed not explicitly enumerating keys that we're not explicitly saving.

**Comment by avrohomgottlieb** [comment]
@davidsmejia I think that we should discuss a more complete testing strategy, specifically to break up these current methods into tests which are more descriptive and meaningful. Also, this testing suite doesn't cover `SCPCP999992`.

File: api/scpca_portal/test/management/commands/test_load_metadata.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+
+class TestLoadMetadata(TransactionTestCase):
+    def setUp(self):
+        self.load_metadata = partial(call_command, "load_metadata")
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        shutil.rmtree(common.OUTPUT_DATA_PATH, ignore_errors=True)
```
Comment (avrohomgottlieb) [inline api/scpca_portal/test/management/commands/test_load_metadata.py:23]: This is not necessary.

**Comment by avrohomgottlieb** [comment]
The tests should assess the permutations of the passed options, making sure that they work as expected, as well as doing general integration testing. Most of the underlying functionality will be unit tested in the `loader` module.

File: api/scpca_portal/management/commands/load_metadata.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    ) -> None:
+        """Loads metadata from input metadata files on s3 and creates model objects in the db."""
+        # Prepare data input directory.
+        common.INPUT_DATA_PATH.mkdir(exist_ok=True, parents=True)
+
+        s3.download_input_metadata(input_bucket_name)
+
+        projects_metadata = metadata_file.load_projects_metadata(
+            Project.get_input_project_metadata_file_path(), scpca_project_id
+        )
```
Comment (davidsmejia) [inline api/scpca_portal/management/commands/load_metadata.py:123]: 1. lets remove the first argument of this function and clean up if we can
2. lets force a kwarg on what is currently called `scpca_project_id` and make it `filter_on_project_id=scpca_project_id`

File: api/scpca_portal/management/commands/load_metadata.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        # Projects can only be intentionally purged.
+        # If the reload_existing flag is not set, then the project should not be procssed.
+        if not reload_existing:
+            logger.info(f"'{project}' already exists. Use --reload-existing to re-import.")
+            return False
+
+        # Purge existing projects so they can be re-added.
+        logger.info(f"Purging '{project}")
+        project.purge(delete_from_s3=update_s3)
+        return True
```
Comment (davidsmejia) [inline api/scpca_portal/management/commands/load_metadata.py:99]: match what you did with `can_process_project` here and make `can_purge_project`

File: api/scpca_portal/management/commands/load_metadata.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                continue
+
+            # If project exists and cannot be purged, then throw a warning
+            project_id = project_metadata["scpca_project_id"]
+            if project := Project.objects.filter(scpca_id=project_id).first():
+                # If there's a problem purging an existing project, then don't process it
+                if not self.purge_project(
+                    project, reload_existing=reload_existing, update_s3=update_s3
+                ):
+                    continue
```
Comment (davidsmejia) [inline api/scpca_portal/management/commands/load_metadata.py:135]: if you `can_purge_project`, then do it here

File: api/scpca_portal/management/commands/load_metadata.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+            project.load_metadata()
+            if samples_count := project.samples.count():
+                logger.info(
+                    f"Created {samples_count} sample{pluralize(samples_count)} for '{project}'"
+                )
+
+        if clean_up_input_data:
+            logger.info("Cleaning up input data")
+            self.clean_up_input_data()
```
Comment (avrohomgottlieb) [inline api/scpca_portal/management/commands/load_metadata.py:154]: Let's put back this back in the loop to do on a per project basis

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**